ramalama/model.py

```diff
@@ -100,6 +99,18 @@ def run(self, args):
         if not args.ARGS:
             exec_args.append("-cnv")
 
+        if args.gpu:
+            if sys.platform == "darwin":
+                # llama.cpp will default to the Metal backend on macOS, so we don't need
+                # any additional arguments.
+                pass
+            elif sys.platform == "linux" and Path("/dev/dri").exists():
```

Comment:
@rhatdan and @slp we need to account for these conditionals in the quadlet generator somehow

Reply:
It is always linux and -/dev/dri will decide to add or not the device to the container.

But not sure how we figure that part out.

Reply:
What happens on linux if  -ngl 99 is added and /dev/dri does not exists.

Reply:
llama.cpp effectively ignores -ngl 99 if it cannot see any available GPU AFAIK

Reply:
The thing is... As I can tell @slp knows... It's hard to justify automatically using a vulkan device if available, because if say you are on a device with simply integrated graphics with a lot of people are, this kind of user is much better off doing CPU inferencing (with -ngl 99 off).

It's why in the ROCm PR I do a check to see if the device has > 1G VRAM